docarray/array/storage/redis/seqlike.py

```diff
@@ -59,6 +60,17 @@ def _upload_batch(self, batch_of_docs: Iterable['Document']):
 
     def _extend(self, docs: Iterable['Document']):
         da = DocumentArray(docs)
+
+        if (
+            self._config.root_id
+            and self._config.index_name.find('_subindex_') != -1
+            and not all([doc.tags.get('root_id', None) for doc in da])
+        ):
+            warnings.warn(
+                "Not all documents have root_id set. This may cause unexpected behavior.",
+                UserWarning,
+            )
+
```

Comment:
If think to handle this properly, and for all backends at once, this should be moved into `extend()` of the base `seqlike`.
If I understand correctly, the challenge here is to detect if a subindex is extended directly. To make this generally possible we might have to introduce a flag `._is_subindex` or similar; I think that is the better trade-off than doing this check in all backends separately